monitor.cpp

```diff
@@ -1,38 +1,115 @@
 #include "./monitor.h"
-#include <assert.h>
-#include <thread>
-#include <chrono>
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+#include <chrono>
+#include <thread>
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+using std::cout, std::flush;
+using std::chrono::seconds;
+using std::this_thread::sleep_for;
+
+enum VitalCondition {
+    BELOW_RANGE,
+    NEAR_LOW,
+    WITHIN_RANGE,
+    NEAR_HIGH,
+    ABOVE_RANGE
+};
+
+struct VitalParameters {
+    float lowerLimit;
+    float upperLimit;
+    float warningThresholdPercent;
+    const char* vitalSign;
+    const char* criticalAlert;
+    const char* lowWarning;
+    const char* highWarning;
+};
+
+float calculateTolerance(const VitalParameters& params) {
+    return params.upperLimit * (params.warningThresholdPercent / 100.0f);
+}
+
+VitalCondition checkLowZone(float reading, float lowerLimit, float nearLowLimit) {
+    if (reading < lowerLimit) return BELOW_RANGE;
+    if (reading < nearLowLimit) return NEAR_LOW;
+    return WITHIN_RANGE;
+}
```

Comment:
Good separation of low from high - it resolves complexity, as well as distinguishes different levels of attention needed